abci/types/pubkey.go

```diff
@@ -3,17 +3,29 @@ package types
 import (
 	"github.com/tendermint/tendermint/crypto/ed25519"
 	cryptoenc "github.com/tendermint/tendermint/crypto/encoding"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+	tmcrypto "github.com/tendermint/tendermint/proto/tendermint/crypto"
 )
 
-const (
-	PubKeyEd25519 = "ed25519"
-)
+func UpdateValidator(pk []byte, power int64, keyType string) ValidatorUpdate {
```

Comment:
Do we have to break the API here? Maybe we can just add a new function instead?

Reply:
yup can do this.